packages/hdwallet-ledger/src/solana.ts

```diff
@@ -0,0 +1,31 @@
+import * as core from "@shapeshiftoss/hdwallet-core";
+import * as bs58 from "bs58";
+
+import { LedgerTransport } from "./transport";
+import { handleError } from "./utils";
+
+const HARDENED = 0x80000000;
+
+// This ensures we generate a derivation compatible with both hw-app-solana (stripped m/) and Phantom (added hardened change component)
+// Using good ol' core.bip32ToAddressNList here = problems
+// https://help.phantom.com/hc/en-us/articles/12988493966227-What-derivation-paths-does-Phantom-wallet-support#h_01HFBRSYQKC73Z43HPKKW0C1SS
+// https://www.npmjs.com/package/@ledgerhq/hw-app-solana#getaddress
+const addressNListToSolanaDerivationPath = (addressNList: number[]): string => {
+  const solanaComponents = addressNList.slice(0, 3);
+
+  const basePath = solanaComponents.map((num) => (num >= HARDENED ? `${num - HARDENED}'` : num)).join("/");
+
+  return `${basePath}/0'`;
+};
+
```

Comment:
Note to reviewers: run app with `yarn dev:web:linked`, put a breakpoint here, and if you do notice account 1+ look odd, it's actually fine.
This can be verified by comparing the bip32 path for account 1+ over the Phantom path as per [Phantom docs ](https://help.phantom.com/hc/en-us/articles/12988493966227-What-derivation-paths-does-Phantom-wallet-support#h_01HFBRSYQKC73Z43HPKKW0C1SS).
We use `m/44'/501'/0'/0'` for account 0 as Phantom does, and `m/44'/501'/1'/0` for account 1 as Phantom does (yay, wallet honouring account_index vs. address_index), however, there is some weird discovery stuff and grouping of addresses across chains happening, so Phantom may be deriving something different as us for those accounts 1+.

Reply:
![image](https://github.com/user-attachments/assets/c70368e3-329b-4bed-bc87-9bc48fd5cb4c)

Seems like phantom, and autodiscovery find the first addy 0 while others addies differs as per 

![image](https://github.com/user-attachments/assets/a663aeea-7c6c-42a3-85e5-dd594501e687)

derivation looks the same as per phantom docs but other addies are different as you documented